### DISCUSSION TITLE
Whisper running in client side javascript

### DISCUSSION THREAD

**Discussion by j0rd1smit** [openai/whisper#1153] [Show and tell]
Using [transformers.js](https://github.com/xenova/transformers.js), ONNX.js, and web assembly, I have made a small demo for Whisper that runs fully on **client-side Javascript**. You can do the following in the demo application:
- Transcribe a video/audio file.
- If you click on the transcribed text, the video/audio play will jump to the timestamp of clicked text.
- You can export the transcription to CSV.
- The video/audio player highlights the transcription of what is currently being said.

You can check it out [over here](https://jordismit.com/tools/ai-based-transcription-in-the-browser/).
![Screenshot 2023-03-27 at 09 38 50](https://user-images.githubusercontent.com/22663455/227873349-d986b160-199d-4c5d-a180-5ad132f5c243.jpg)

Source code:
- [whisper-based-transcription.js](https://github.com/j0rd1smit/j0rd1smit.github.io/blob/main/assets/js/whisper-based-transcription.js)
- [whisper.worker.js](https://github.com/j0rd1smit/j0rd1smit.github.io/blob/main/assets/js/whisper.worker.js)

**Comment by miwgel** [comment]
That is super cool

**Comment by rajeevparshad** [comment]
How to stop auto correction
